baseplate/sidecars/live_data_watcher.py

```diff
@@ -50,8 +89,20 @@ def on_change(self, data: bytes, _znode_stat: ZnodeStat) -> None:
             if self.owner and self.group:
                 os.fchown(tmpfile.fileno(), self.owner, self.group)
             os.fchmod(tmpfile.fileno(), self.mode)
-
-            tmpfile.write(data)
+            try:
+                json_data = json.loads(data.decode("UTF-8"))
+            except json.decoder.JSONDecodeError:
+                # If JSON fails to decode, still write the bytes data since
+                # we don't necessarily know if the the contents of the znode
+                # had to be valid JSON anyways.
+                tmpfile.write(data)
+            else:
+                # If no exceptions, we have valid JSON, and can parse accordingly.
+                data_to_write = NodeWatcher.get_data_to_write(json_data, data)
+                if data_to_write is not None:
+                    tmpfile.write(data_to_write)
+                else:
+                    logger.warning("No data written to destination node.")
```

Comment:
This would only occur in the current scenario if 'live_data_watcher_load_type' was 'http', but there was no URL or the data from the URL was not fetchable.

Otherwise, it'll always fallback to `tmpfile.write(data)` where data is just whatever was in the ZK source.